ratpack-core/src/main/java/ratpack/http/client/internal/DefaultHttpClient.java

```diff
@@ -88,6 +95,10 @@ private DefaultHttpClient(DefaultHttpClient.Spec spec) {
     } else {
       proxy = null;
     }
+    if (spec.resolver == null) {
+      LOGGER.warn("DNS resolver not specified for HttpClientSpec, defaulting to JVM blocking implementation");
+      spec.resolver = HttpClient.blockingResolver();
```

Comment:
Do the only sane thing we can if someone is creating a new HttpClient and doesn't provide a resolver, which is use the blocking implementation but WARN about it.

We can't default to a non-blocking implementation because we don't have a handle to an `EventLoop` which is needed. We could defer to to the point that we make the request but that would create high GC churn and possible performance impacts as each request would create its own resolver.